docs/developer/develop-vscode.md

```diff
@@ -0,0 +1,135 @@
+# Developing with Visual Studio Code (no cloud required)
+
+## Introduction
+This procedure will get you up and running with a Visual Studio Code environment for Turbinia development. The provided configuration files will create a development container containing all dependencies, pylint and yapf correctly setup and launch configurations for both client, server and workers. With this setup it is possible to initiate full Turbinia debug sessions including breakpoints, watches and stepping.
+
+You can set Visual Studio Code up to run a local stack (using redis and celery) or use a hybrid GCP stack (using pubsub, datastore and cloud functions). We advice you to run a local stack if you don't need to debug or develop Turbinia GCP functionality.
+
+## Before you start
+*   Check out the [How it Works](../user/how-it-works.md) page to see how the different
+    components work within Turbinia.
+*   Make sure to follow the Turbinia
+    [developer contribution guide](contributing.md).
+
+#### Step 1 - Install required software
+Prepare your OS:
+* Install Visual Studio Code and install the Remote Development extension pack.
+* Install Docker on your operating system (eg Docker Desktop on OSX)
+
+#### Step 2 - Fork Turbinia
+Fork Turbinia on Github and create a new feature branch to work on.
+```
+$ git clone https://github.com/[your-github-user-id]/turbinia.git
+$ git remote add upstream https://github.com/google/turbinia.git
+$ git checkout -b my-new-feature
+$ cd turbinia
+```
+
+#### Step 3 - Open in Visual Studio Code
+Open the folder in vscode and choose to "Reopen in Container" when asked (vscode will see the ```.devcontainer``` folder in the turbinia cloned source tree). Vscode will build your Turbinia development container and that will take a couple of minutes.
+
+Note: if vscode does not ask you to reopen in a container you need to verify you have installed the Remote Development extension!
+
+When this is finished continue with Step 4 for a local Turbinia setup or Step 6 for a GCP hybrid setup. 
+
+#### Step 4 - Local Turbinia setup
+The local turbinia setup will use redis and celery. Let's create the configuration file for this setup.
+```
+$ sed -f ./docker/vscode/redis-config.sed ./turbinia/config/turbinia_config_tmpl.py > ~/.turbiniarc
+```
+
+#### Step 5 - Local Turbinia verification
+Let's verify that the local setup is working before we start developing and debugging. We are going to start a server and worker in seperate vscode terminals and create a Turbinia request in a third.. Open up 3 vscode terminals and execute below commands.
+
+Start server
+```
+$ python3 turbinia/turbiniactl.py -S server
+```
+Start worker
+```
+$ python3 turbinia/turbiniactl.py -S celeryworker
+```
+Fetch and process some evidence
```

Comment:
Maybe mention that the request ID is the in the output of the first turbiniactl.py command?